yew/src/lib.rs

```diff
@@ -97,9 +97,101 @@
 extern crate self as yew;
 
 /// This macro implements JSX-like templates.
+///
+/// This macro erases the component type and always returns [`Html`].
+/// If you want non-erased type, look at [`html_nested!`] macro.
+///
+/// More information about using the `html!` macro can be found in yew's [manual].
+///
+/// [`Html`]: ./html/type.Html.html
+/// [`html_nested!`]: ./macro.html_nested.html
+/// [manual]: https://yew.rs/docs/en/concepts/html/
 pub use yew_macro::html;
 
-#[doc(hidden)]
+/// This macro is similar to [`html!`], but returns actual component type instead
+/// of a generic [`Html`].
+///
+/// That macro is useful when, for example, in a typical implementation of a list
+/// component (let's assume it's called `List`). In a typical implementation you
+/// would have two connected components -- `List` and `ListItem`, and `List`'s
+/// children would be a number of `ListItem`s.
+///
+/// You can find an example implementation of this in the [`nested_list`] example.
+/// That example shows, how to create static lists with their children, but in
+/// most use cases the contents of a list is dynamic and thus known only at runtime.
```

Comment:
I think we should update the `nested_list` example to include dynamically generated children too. Would you mind creating an issue for this, @Mingun?

```suggestion
/// That example shows, how to create static lists with their children.
```

Reply:
https://github.com/yewstack/yew/issues/1534